### PR TITLE
Really update the profile promised by addToTypeIndex

### PR DESCRIPTION
**Pull request by dan-f**
@dmitrizagidulin please review this

### CODE REVIEW

File: src/util/graph-util.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ function statementToNT (statement, excludeDot) {
 function graphFromStatements (statements, rdf) {
   var graph = rdf.graph()
   statements.forEach(function (st) {
-    graph.addStatement(st)
+    graph.add(st)
```
Comment (dan-f) [inline src/util/graph-util.js]: This was bizarre. The `addStatement` method was not updating the indices and breaking querying. Will report that bug.

File: test/unit/type-registry-test.js [hunk 2]
```diff
@@ -104,3 +106,25 @@ test('SolidProfile addTypeRegistry() test', function (t) {
 
   t.end()
 })
+
+test('type registry addToTypeIndex() updates the profile with new registry when there was previously no type index', t => {
+  nock('https://localhost:8443/')
```
Comment (dan-f) [inline test/unit/type-registry-test.js]: Nock is a nifty library for declaratively mocking out and responding to node's http requests.

**Comment by dmitrizagidulin** [comment]
Looks good! (Also, I suspect we'll need to apply the same fix in [app-registry.js](https://github.com/solid/solid-client/blob/24018400dc8955677b0b30f8eb2c2823338e6677/src/app-registry.js#L60))

**Comment by dan-f** [comment]
Ok, I may as well fix that while I'm in here. I find it curious that we have two modules with the all the same exported functions but with "app" vs "type" in the names. Could we have not shared logic?

**Comment by dmitrizagidulin** [comment]
Thanks!

As for two very similar registries - yeah, it could certainly be refactored to share the logic. I don't think it's a high priority, though. (Maybe if we add another registry type, it would be worth it.)

**Comment by dan-f** [comment]
My point is just that it's technical debt to be required to make the same code change in two files. Plus without your institutional knowledge of knowing about the app registry module I would have just merged this fix and there would have been a bug remaining in the app registry

`¯\_(ツ)_/¯`

**Comment by dmitrizagidulin** [comment]
You're right, of course. The reason I think it's not worth to tackle this corner of tech debt yet is - we don't really use the app registry for anything, at the moment.

**Comment by dan-f** [comment]
Totally

**Comment by dan-f** [comment]
@dmitrizagidulin I made the same fix to the app registry

**Comment by dmitrizagidulin** [comment]
Thanks! :) Good to go.

**Pull request merged**